xds/internal/xdsclient/pool.go

```
@@ -144,6 +144,13 @@ func (p *Pool) SetFallbackBootstrapConfig(config *bootstrap.Config) {
 	p.config = config
```

Given that the documentation says this is used to specify a bootstrap configuration that will be used as a fallback when the bootstrap environment variables are not defined, should we check for `p.config != nil` before writing to it?

---

I checked C++, and it does look like the fallback config should only be used when the env vars are not set. So, we do need a check here.

---

And also, we should probably throw an error log in that condition, so that the user is aware that they have set the env vars *and* they are also trying to set the fallback bootstrap configuration.

---

Added the error log if already set

---

We don't run any tests of the same package in parallel. A grep for `t.Parallel` returns no results in our repo.

```
easwars@minetop: grpc-go$ grep -r "t.Parallel" *
easwars@minetop: grpc-go$ 
```

Tests from different packages are automatically run in parallel by the `go test` command though. So, if you are seeing problems with this, then some other tests also use the global default pool. Can you please check? Thanks.

---

So, csds test and one of the c2p resolver test uses default xds client pool